x-pack/legacy/plugins/ml/public/datavisualizer/index_based/page.tsx

```diff
@@ -166,11 +168,14 @@ export const Page: FC = () => {
   const [nonMetricFieldQuery, setNonMetricFieldQuery] = useState(defaults.nonMetricFieldQuery);
 
   useEffect(() => {
-    const timeUpdateSubscription = timefilter.getTimeUpdate$().subscribe(loadOverallStats);
-    return () => {
-      timeUpdateSubscription.unsubscribe();
+    const subscriptions = new Subscription();
+    subscriptions.add(timefilter.getTimeUpdate$().subscribe(() => loadOverallStats()));
```

Comment:
nit:
```suggestion
import {merge} from 'rxjs';
...
merge(timefilter.getTimeUpdate$(),  mlTimefilterRefresh$).subscribe(loadOverallStats)
```

Reply:
Nice suggestion to use `merge`. Implemented in https://github.com/elastic/kibana/pull/46708/commits/7dcb5824e374574fd294594752bfdc7c9aecbdfb